### PR TITLE
Tests parallel setup

### PR DESCRIPTION
**Pull request by i1i1**
https://jira.hyperledger.org/browse/IR-1009

Signed-off-by: i1i1 <vanyarybin1@live.ru>

### Description of the Change

This commit enables parallel testing for iroha_client crate

### Benefits

Tests will be a lot faster and more error prune

### CODE REVIEW

File: iroha_http_server/src/lib.rs [hunk 1]
```diff
@@ -42,50 +42,57 @@ impl<State: Clone + Send + Sync + 'static> Server<State> {
         }
     }
 
+    async fn start_handle(
```
Comment (eadventurous) [inline iroha_http_server/src/lib.rs:45]: Can you add some docs here?

File: iroha_client/tests/tests/tests.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+mod account_with_multiple_signatories;
```
Comment (eadventurous) [inline iroha_client/tests/tests/tests.rs:1]: Wow `.../tests/tests/tests.rs` does this need to be this much nested?

File: iroha_http_server/src/lib.rs [hunk 3]
```diff
@@ -42,50 +42,57 @@ impl<State: Clone + Send + Sync + 'static> Server<State> {
...
     /// Starts server at `address`
     ///
     /// # Errors
     /// Fails if accepting one of client fails
     pub async fn start(&self, address: &str) -> iroha_error::Result<()> {
         let listener = TcpListener::bind(address).await?;
         loop {
-            let (mut stream, _) = match listener.accept().await {
+            let (stream, _) = match listener.accept().await {
                 Ok(ok) => ok,
```
Comment (eadventurous) [inline iroha_http_server/src/lib.rs:87]:
```suggestion
                Ok(stream) => stream,
```

File: iroha_client/tests/tests/tests.rs (hunk 2 above)
Comment (i1i1) [reply iroha_client/tests/tests/tests.rs:1]: Unfortunately. May be we can rename it but I didn't come up with good name

Comment (i1i1) [reply iroha_client/tests/tests/tests.rs:1]: We need it as `tests/*.rs` files are treated by rust as separate binaries. What is why we need to have only 1 file in `tests` directory and many files nested.

Comment (eadventurous) [reply iroha_client/tests/tests/tests.rs:1]: But shouldn't it be then `iroha_client/tests/tests.rs`
And other files in `iroha_client/tests/tests/..`?

Comment (eadventurous) [reply iroha_client/tests/tests/tests.rs:1]: Ok, I see.

**Review by eadventurous — APPROVED** [review]

**Pull request merged**